thanks/thanks.py

```diff
@@ -1,82 +1,97 @@
-                        colored('enjoy donations!', 'green'),
-                    ), attrs=['bold'])
-                table_data = []
-                for contributor in self.give_thanks_to:
-                    table_data.append(
-                        [
-                            colored(contributor, 'cyan'),
-                            colored(self.give_thanks_to[contributor]['url'], 'green'),
-                            colored(", ".join(set(self.give_thanks_to[contributor]['packages'])), 'red'),
-                        ]
-                    )
-                print(format_pretty_table(
-                    data=table_data,
-                    column_names=['Author', 'Where to thank', 'Packages'],
-                    horizontal_bar=' ',
-                    vertical_bar=' ',
-                ))
+    def load_local_project_data(self):
+        print('Loading data about {}'.format(colored('contributors...', 'cyan')))
+        with open(JSON_FILE, 'r') as fh:
+            data = json.load(fh)
+            for project_name in data.keys():
+                authors = ', '.join(data[project_name].get('authors', []))
+                data[project_name]['authors'] = authors
+        return data
+
+    def find_project_details(self, requirements_list):
+        print('Scanning your {} file...'.format(colored('requirements', 'red')))
+        reqs = [
+            next(requirements.parse(r))
+            for r in requirements_list
+        ]
+        for req in reqs:
+            self.update_give_thanks(req.name)
+        self._display_thanks()
+
+    def _display_thanks(self):
+        if self.give_thanks_to:
+            cprint(
+                'You depend on {} who would {}'.format(
+                    colored('{} authors'.format(len(self.give_thanks_to)), 'cyan'),
+                    colored('enjoy donations!', 'green'),
+                ), attrs=['bold'])
+            colorized_data = [
+                ProjectData(
+                    name=colored(pd.name, 'cyan'),
+                    funding_link=colored(pd.funding_link, 'green'),
+                    authors=colored(pd.authors, 'yellow'),
+                )
+                for pd in self.give_thanks_to.values()
+            ]
+
+            print(format_pretty_table(
+                data=colorized_data,
+                column_names=['Project', 'Where to thank', 'Authors'],
+                horizontal_bar=' ',
+                vertical_bar=' ',
+            ))
+
+    def get_local_data(self, project_name):
+        if project_name in self.data:
+            data = ProjectData(
+                name=project_name,
+                funding_link=self.data[project_name]['url'],
+                authors=self.data[project_name]['authors']
+            )
+        else:
+            data = None
+        return data
+
+    def get_remote_data(self, project_name):
+        url_format = 'https://pypi.org/pypi/{}/json'.format
```

Comment:
Awesome. I never wanted to use the xmlrpc api, but I don't think the json was available in the old cheeseshop.